x/twap/logic_test.go

```diff
@@ -1596,3 +1596,129 @@ func geometricTestCaseFromDeltas0(startAccum, accumDiff sdk.Dec, timeDelta time.
+			priceValue: osmomath.MustNewDecFromStr("912648174127941279170121098210.928219201902041311"),
+			// From: https://www.wolframalpha.com/input?i2d=true&i=log+base+2+of+912648174127941279170121098210.928219201902041311+with+20+digits
+			expected: osmomath.MustNewDecFromStr("99.525973560175362367"),
+		},
+	}
+	for name, test := range tests {
+		s.Run(name, func() {
+			osmoassert.ConditionalPanic(s.T(), test.expectedPanic, func() {
+				result := twap.TwapLog(test.priceValue.SDKDec())
+				require.True(osmomath.DecApproxEq(s.T(), test.expected, osmomath.BigDecFromSDKDec(result), expectedErrTolerance))
+			})
+		})
+	}
+}
+
+func (s *TestSuite) TestTwapPow() {
+	var expectedErrTolerance = osmomath.MustNewDecFromStr("0.00000100")
+
+	tests := map[string]struct {
+		exponentValue osmomath.BigDec
+		expected      osmomath.BigDec
+		expectedPanic bool
+	}{
+		"TwapPow(0) = 1": {
+			exponentValue: osmomath.ZeroDec(),
+			expected:      osmomath.OneDec(),
+		},
+		"TwapPow(0.1) = 1.071773": {
+			exponentValue: osmomath.MustNewDecFromStr("0.1"),
+			//From: https://www.wolframalpha.com/input?i2d=true&i=power+base+2+exponent+0.1+with+7+digits
+			expected: osmomath.MustNewDecFromStr("1.071773"),
+		},
+		"TwapPow(0.12345678) = 1.08934186": {
+			exponentValue: osmomath.MustNewDecFromStr("0.12345678"),
+			//From: https://www.wolframalpha.com/input?i2d=true&i=power+base+2+exponent+0.12345678+with+9+digits
+			expected: osmomath.MustNewDecFromStr("1.08934186"),
+		},
+		"TwapPow(0.5) = 1.41421356": {
+			exponentValue: osmomath.MustNewDecFromStr("0.5"),
+			//From: https://www.wolframalpha.com/input?i2d=true&i=power+base+2+exponent+0.5+with+9+digits
+			expected: osmomath.MustNewDecFromStr("1.41421356"),
+		},
+		"TwapPow(0.7654321) = 1.699879": {
+			exponentValue: osmomath.MustNewDecFromStr("0.7654321"),
+			//From: https://www.wolframalpha.com/input?i2d=true&i=power+base+2+exponent+0.7654321+with+7+digits
+			expected: osmomath.MustNewDecFromStr("1.699879"),
+		},
+		"TwapPow(0.9) = 1.86606598": {
+			exponentValue: osmomath.MustNewDecFromStr("0.9"),
+			//From: https://www.wolframalpha.com/input?i2d=true&i=power+base+2+exponent+0.9+with+9+digits
+			expected: osmomath.MustNewDecFromStr("1.86606598"),
+		},
+		"TwapPow(0.99999999) = 1.99999999": {
+			exponentValue: osmomath.MustNewDecFromStr("0.99999999"),
+			//From: https://www.wolframalpha.com/input?i2d=true&i=power+base+2+exponent+0.99999999+with+9+digits
+			expected: osmomath.MustNewDecFromStr("1.99999999"),
+		},
+		"TwapPow(1); invalid; panic": {
+			exponentValue: osmomath.OneDec(),
+			expectedPanic: true,
+		},
+	}
+	for name, test := range tests {
+		s.Run(name, func() {
+			osmoassert.ConditionalPanic(s.T(), test.expectedPanic, func() {
+				result := twap.TwapPow(test.exponentValue.SDKDec())
+				require.True(osmomath.DecApproxEq(s.T(), test.expected, osmomath.BigDecFromSDKDec(result), expectedErrTolerance))
```

Comment:
Similar suggestion for the log test above
```suggestion
				suite.Require().True(test.expected, osmomath.BigDecFromSDKDec(result), expectedErrTolerance))
```

Reply:
Can you please apply these changes everywhere where applicable?

Reply:
Yeah, I'm understanding the code inside to be able to change according to your suggestion is like this, right?
`s.Require().True(test.expected.Sub(osmomath.BigDecFromSDKDec(result)).Abs().LTE(expectedErrTolerance))`
or I can write another func for this test's type because `osmomath.DecApproxEq` is not being used for `suite`, tell me if I am wrong something, thanks!

Reply:
The current change LGTM!